docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
+});
 
-  job.run()
-})
+events.process();
 ```
-[brigade-16.js](../../examples/brigade-16.js)
-
-This script creates a new job and then enables the cache. Then it runs two different
-tasks. The first writes the build's unique ID into a file in the cache, and the second one
-echos the contents of that generated file.
-
-If we run the above a few times and then check the job's output, we'll see one ID for each
-time the job was run.
+[08-shared-workspace](https://github.com/brigadecore/brigade/tree/v2/examples/08-shared-workspace)
 
 ```console
-$ kubectl logs cacher-1504091963651-master
-brigade-worker-01brzvq79mj4849vy7ae0fez44-master
-brigade-worker-01brzvqnhz21yjjh8m0zyxrxsc-master
-brigade-worker-01brzvrhwe50xtktf4gcqk94wj-master
-```
-
-This happens because each time the job runs, the new build ID is written into the same
-file that the job used on other builds.
-
-By default, job caches are limited to only 5 megabytes (`5Mi`). However, you can easily
-change this by setting `job.cache.size` to a larger value (`50Mi`, `5Gi`).
-
-There are two final observations to make about job caches:
+$ brig event create --project shared-workspace
 
-1. While a job cache is designed to persist across multiple runs, they are still
-   considered to be _volatile storage_, which means a cache can reset. Do not use it as if
-   it were stable long-term storage.
-2. Caches are not automatically destroyed by Brigade (though other systems may clean them).
-   That means that if you add lots and lots of jobs with caches enabled, lots of storage
-   space will be reserved even if it is unused.
+Created event "2eee9044-4469-49bd-a58b-aa659951a502".
 
-### Docker Runtime
+$ brig event logs --id 2eee9044-4469-49bd-a58b-aa659951a502 --job second-job
 
-Each job has the option to mount in a docker socket. When enabled, a docker socket is mounted to
-`/var/run/docker.sock` in the job's container.
-
-In order for the socket to be mounted, the Brigade project must have `Allow host mounts` set to
-`true`. This can be set via the Advanced Configuration Options during `brig create`.  (See the [projects](projects.md)
-doc for further info.)
-
-This is a toggle-able option for all jobs, but is not enabled by default. A job must declare that it
-needs a docker socket.
+Hello!
+```
 
-```javascript
-const { events, Job } = require("brigadier")
+> Note: Shared storage is dependent on the underlying Kubernetes cluster and
+> the availability of the correct type of storage class. See the [Storage] doc
+> for more information on cluster requirements.
 
-events.on("exec", (e, p) => {
-  var one = new Job("one", "docker:17.06.0-ce", ["docker images"])
+[Storage]: /topics/operators/storage
 
-  one.docker.enabled = true
+## Sidecar containers
 
-  one.run()
-})
-```
+Jobs can optionally be configured with one or more sidecar containers, which
+run alongside the job's primary container. All sidecar containers will be
+terminated a short time after the job's primary container completes.
```

Comment:
Should we mention somewhere that the success/failure of a job is contingent only on the exit code of the primary container and is not (directly) affected by the sidecar containers?

Reply:
Some other things that we probably need to note in this section:

1. All the containers are networked together such than processes listening for network connections in any one of them can be addressed by processes running in the others using the local network interface.

1. Brigade does not understand the relationship(s) between your containers and if, for instance, a process in the primary container should be delayed until some supplementary process in some sidecar container is up and running and listening for connections, then the script author needs to account for that themsevles.